.github/workflows/java.yml

```diff
@@ -23,30 +23,49 @@ jobs:
             filename: codemp.dll
           - runner: macos-14
             target: darwin-arm64
-            filename: codemp.dylib
+            filename: libcodemp.dylib
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
       - uses: arduino/setup-protoc@v3
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - run: cargo build --release --features=java
+      - uses: actions/upload-artifact@v4
+        with:
+          name: codemp-java-${{ matrix.platform.target }}
+          path: target/release/${{ matrix.platform.filename }}
+
+  publish:
+    needs: [build]
+    runs-on: ubuntu-latest 
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          path: dist/java/artifacts
+          pattern: codemp-java-*
+          merge-multiple: true
+      - run: tree
+        working-directory: dist/java
       - uses: actions/setup-java@v4
         with:
             distribution: 'temurin'
             java-version: '11'
       - uses: gradle/actions/setup-gradle@v4
         with:
-          gradle-version: "8.10" # Quotes required to prevent YAML converting to number
-      - run: gradle build
+          gradle-version: "8.10"
+      - run: gradle windowsJar
+        working-directory: dist/java
+      - run: gradle macosJar
+        working-directory: dist/java
+      - run: gradle linuxJar
```

Comment:
```suggestion
```
i dont think these are needed, can we remove them to make the publishing process faster?